pkg/render/opt.go

```diff
@@ -0,0 +1,50 @@
+package render
+
+import "fmt"
+
+type TplOptions struct {
+	MissKeyStrategy MissKeyStrategy
+}
+
+func NewOptions(opts ...TplOption) *TplOptions {
+	o := &TplOptions{}
+	for _, opt := range opts {
+		opt(o)
+	}
+	return o
+}
+
+type TplOption func(options *TplOptions)
+
+func WithMissKeyStrategy(st MissKeyStrategy) TplOption {
+	return func(options *TplOptions) {
+		options.MissKeyStrategy = st
+	}
+}
+
+type MissKeyStrategy string
+
+const (
+	// MissKeyStrategyDefault The default behavior: Do nothing and continue execution.
+	//		If printed, the result of the index operation is the string
+	//		"<no value>".
+	MissKeyStrategyDefault MissKeyStrategy = "default"
+	// MissKeyStrategyInvalid = MissKeyStrategyDefault
+	MissKeyStrategyInvalid MissKeyStrategy = "invalid"
+	// MissKeyStrategyZero  returns the zero value for the map type's element.
+	// returns <no value> if use map[string]interface{}
+	MissKeyStrategyZero MissKeyStrategy = "zero"
+	// MissKeyStrategyError Execution stops immediately with an error.
+	MissKeyStrategyError MissKeyStrategy = "error"
```

Comment:
`MissKeyStrategyError ` is well(it is default option, very nice), is here anywhere user could change it?If not, I think we have 2 way to fix:
1. remove the `TplOption`,`MissKeyStrategy `, and set template's option to `missingkey=error`, so we could make code simple and clear(recommend)
2. give a way to configure `MissKeyStrategy`(not recommend, I could not image why user need to configure it)